metasv/generate_sv_intervals.py

```diff
@@ -238,54 +263,65 @@ def get_operator_fn(op):
-    if columns and not orders:
-        print "Bad c and o combinations"
-    
-    if len(columns)>1 and len(orders)==1:
-        orders=[ords for c in columns]
-
-    if len(columns)!=len(orders):
-        print "Bad c and o combinations"
+def blind_merge(intervals,cols,ops):
+    func_logger = logging.getLogger("%s-%s" % (blind_merge.__name__, multiprocessing.current_process()))
+    try:
+        columns=map(lambda x:int(x),cols.split(','))
+        operations=ops.split(',')
+        if columns and not operations:
+            func_logger.error("Aborting!")
+            raise Exception("Bad column and operation combinations for merge: %s, %s\n" % (str(columns), operations))    
+        if len(columns)>1 and len(operations)==1:
+            operations=[ops for c in columns]
+
+        if len(columns)!=len(operations):
+            func_logger.error("Aborting!")
+            raise Exception("Bad column and operation combinations for merge: %s, %s\n" % (str(columns), operations))    
     
-    column_operations={c:orders[i] for i,c in enumerate(columns)}    
+        column_operations={c:operations[i] for i,c in enumerate(columns)}    
     
-    if not intervals:
-        return intervals
+        if not intervals:
+            return intervals
     
-    start = int(intervals[0].start)
-    end = int(intervals[0].end)
-    chrom = intervals[0].chrom
-    other_fields = {c:[intervals[0].fields[c-1]] if c<=len(intervals[0].fields) else [""]  for c,o in column_operations.iteritems()}
-    for interval in intervals[1:]:
-        start = min(start, int(interval.start))
-        end = max(end, int(interval.end))
-        for c in column_operations: 
-            other_fields[c].append(interval.fields[c-1])
-    for c,o in column_operations.iteritems():
-        if o == "collapse":
-            other_fields[c] = ",".join(other_fields[c])
-        elif o in  ["sum", "min", "max"]:
-            other_fields[c] = "%s"%get_operator_fn(o)(map(lambda x:int(x) if x else 0,other_fields[c]))
-        elif o == "first":
-            other_fields[c] = other_fields[c][0]
-        elif o == "last":
-            other_fields[c] = other_fields[c][-1]
-        elif o == "distinct":
-            other_fields[c] = ",".join(set(other_fields[c]))
-        elif o == "count":
-            other_fields[c] = len(other_fields[c])
-        else:
-            print "not supported operation"
+        start = intervals[0].start
+        end = intervals[0].end
+        chrom = intervals[0].chrom
+        other_fields = {c:[intervals[0].fields[c-1]] if c<=len(intervals[0].fields) else [""]  for c,o in column_operations.iteritems()}
+        for interval in intervals[1:]:
+            start = min(start, int(interval.start))
+            end = max(end, int(interval.end))
+            for c in column_operations: 
+                other_fields[c].append(interval.fields[c-1])
+        for c,o in column_operations.iteritems():
+            if o == "collapse":
```

Comment:
might be better to use a dictionary from operation to function and simply apply the selected function to other_fields. will make the code cleaner.